File: Projects/rosalind/PPER.java
import java.io.*;
import java.util.*;

public class PPER {
	public static void main(String args[]) throws IOException {
		Scanner in = new Scanner(System.in);
		final int n = in.nextInt(), k = in.nextInt();
		int ans = 1;
		for (int i = n; i > n-k; i--)
			ans = ans * i % 1000000;
		System.out.println(ans);
		in.close();
	}
}

File: Projects/rosalind/PPER.kt
package rosalind

import java.io.IOException
import java.util.*

object PPER {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        val n = stdin.nextInt()
        val k = stdin.nextInt()
        var ans = 1
        for (i in n downTo n - k + 1) ans = ans * i % 1000000
        println(ans)
        stdin.close()
    }
}